collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2888,6 +2886,34 @@ private void updateFieldListQuestions(FormIndex lastChangedIndex) {
         }
     }
 
+    private void saveAnswersForCurrentScreenOneByOneToSupportCalculations(FormEntryPrompt[] mutableQuestionsBeforeSave, List<ImmutableDisplayableQuestion> immutableQuestionsBeforeSave) {
+        FormController formController = getFormController();
+        ODKView currentView = getCurrentViewIfODKView();
+        if (formController == null || currentView == null) {
+            return;
+        }
+
+        int index = 0;
+        for (Map.Entry<FormIndex, IAnswerData> answer : currentView.getAnswers().entrySet()) {
+            if (isQuestionNotRecalculated(mutableQuestionsBeforeSave[index], immutableQuestionsBeforeSave.get(index))) {
```

Comment:
I have a hard time with the negative in the method name. In fact, you could just use `!Objects.equals` here, right? I think that would make things easier to understand. Then you could have a comment above the condition that says something like `questions with `calculates` will have their answers updated as the questions they depend on are saved`.

Reply:
Fixed.